enhance: macro rendered result has unnecessary right margin [breaking change]

stdword opened the pull request:
I've added styles to leverage one-liner macros. Fixes #8623

|before PR|after PR|
|:--:|:--:|
|<img width="200px" src="https://github.com/logseq/logseq/assets/1984175/b841e9fd-73d3-4a06-91d7-e7f00a0ff23b"/>|<img width="220px" src="https://user-images.githubusercontent.com/1984175/235907263-29a60c10-b0eb-473f-852f-c605065cc09d.png"/>|


**NOTE**: This message is a summary and was updated. In the comments below you can find a history of all the changes during working on this PR.


# Changes:
1) macro alignment in one line (add `.inline` class)
2) macro with headings left intact (no addition of `.inline` class)
3) remove right margin (`.mr-1`)
4) change styles for root div of hiccup & html syntax to «one-line» them too (add `.inline` class)

# Test data
```edn
 :macros {
    "theme-test-macro0" "<b>macro</b>",
    "theme-test-macro1" "[:b \"macro\"]",
    "theme-test-macro2" "macro",
    "theme-test-macro3" "header\nmacro",
    
    "iframe" "<iframe src='$1' style='height:320px'></iframe>",
 }
```

```edn
- before {{theme-test-macro0}} after
- before {{theme-test-macro1}} after
- before {{theme-test-macro2}} after
- before {{theme-test-macro3}} after
- {{iframe https://wikipedia.com}}  ;; iframe should be as wide as Logseq block
```

# Related discussions
- https://discuss.logseq.com/t/macro-for-inline-html-is-rendered-as-block/11466
- https://discuss.logseq.com/t/macros-hiccup-and-block-embeds/13930

# ⚠️ Caveats
- This is my first **_clojure_** PR. I feel myself unsure on how I write this code. Please point me to issues if need be

stdword committed:
fix: macro rendered result has unnecessary right margin

stdword commented:
Well, CI / lint is failed in some other code 🤷‍♂️

stdword committed:
rework without new signature

logseq-cldwalker commented:
Hi. I'm not able to repro the screenshot as a macro like `{{ok foo}} | ok` normally looks like 
<img width="204" alt="Screen Shot 2023-04-26 at 2 42 32 PM" src="https://user-images.githubusercontent.com/97210743/234672641-93521d24-8d3c-4094-8940-2867b40bcab2.png">

Could you provide an example block and any custom.css you have that results in this case?

stdword committed:
style: align macro in one line

stdword commented:
Whooh... you are right. No custom css and theme inductions but there was a plugin that provide styles. Good point for me: plugins can change style too.

I've add `.inline-block` style to leverage one-liner macros in new commit.

So there are two changes here:
1) macro alignment in one line
2) remove right margin

Note, macros with headers still look like without this changes:
<img width="200px" src="https://user-images.githubusercontent.com/1984175/234702186-836fbe78-1f84-4707-8199-9db9f4ade993.png"/>

```edn
 :macros {
    "theme-test-macro1" "|macro|",
    "theme-test-macro2" "header\nmacro|",
 }
```

logseq-cldwalker commented:
Thanks for the instructions.

> macro alignment in one line

This is a breaking change. I don't use macros enough to have an opinion on this so will defer to others on the team

stdword commented:
@logseq-cldwalker sure, thank you 🙏!

Please, consider this discussions:
- https://discuss.logseq.com/t/macro-for-inline-html-is-rendered-as-block/11466
- https://discuss.logseq.com/t/macros-hiccup-and-block-embeds/13930

stdword committed:
fix: return default rendering behavior for macros with headings

stdword commented:
@logseq-cldwalker, update PR to left **default** rendering behavior for macros with headings.
<img width="250px" src="https://user-images.githubusercontent.com/1984175/235907263-29a60c10-b0eb-473f-852f-c605065cc09d.png"/>

So, now it changes only one-liner macros.

PS calling `list` function relies on [this](https://github.com/weavejester/hiccup/wiki/Syntax#expanding-seqs).

stdword committed:
fix: change list to vector

mgsergio commented on src/main/frontend/components/block.cljs:
```diff
@@ -936,23 +936,30 @@
-         (inline-text format macro-content)))
-     [:span.warning {:title (str "Unsupported macro name: " name)}
-      (macro->text name arguments)])])
+  (vec
+   (cons
+    :div.macro
+    (let [attributes {:data-macro-name name}]
```
```clojure
(apply vector
            :div.macro
            (let ...            
```
Or 
```clojure
(into [:div.macro]
         (let ...
```

stdword committed:
refactor: simplify vector merge

stdword commented:
Just found an issue I've fixed in this PR: #8623

stdword committed:
fix: change display style for macros to preserve iframe width

stdword commented:
Found additional case for iframe (described [here](https://github.com/stdword/logseq13-full-house-plugin/issues/22)). So I've changed the display style of macro-div (`.inline` instead of `.inline-block`). But Hiccup & HTML syntax inside macro gives an additional div block, so I've added styles for it to be inlined too.

Updated first post to keep all summary information about changes. This comment is for history reasons.

stdword commented:
@tiensonqin did you accidentally remove commits from PR?
<img width="600px" src="https://github.com/logseq/logseq/assets/1984175/4fb3f9f7-e74f-4152-9bc9-303e9f50e6de"/>
